src/bin/ayb.rs

```diff
@@ -169,174 +182,204 @@ async fn main() -> std::io::Result<()> {
-        if let Some(url) = matches.get_one::<String>("url") {
-            let client = AybClient {
-                base_url: url.to_string(),
-                api_token: matches.get_one::<String>("token").cloned(),
-            };
-            if let Some(matches) = matches.subcommand_matches("create_database") {
-                if let (Some(entity_database), Some(db_type)) = (
-                    matches.get_one::<EntityDatabasePath>("database"),
-                    matches.get_one::<DBType>("type"),
-                ) {
-                    match client
-                        .create_database(
-                            &entity_database.entity,
-                            &entity_database.database,
-                            db_type,
-                        )
-                        .await
-                    {
-                        Ok(_response) => {
-                            println!(
-                                "Successfully created {}/{}",
-                                entity_database.entity, entity_database.database
-                            );
-                        }
-                        Err(err) => {
-                            println!("Error: {}", err);
-                        }
+        let config_path = if let Some(path) = matches.get_one::<PathBuf>("config") {
+            path.clone()
+        } else {
+            ProjectDirs::from("org", "ayb", "ayb")
+                .expect("can't determine ayb project directory directory")
+                .config_dir()
+                .join("ayb.json")
+        };
+        let mut config = ClientConfig::from_file(&config_path)?;
+
+        if let Some(matches) = matches.subcommand_matches("set_default_url") {
+            if let Some(url) = matches.get_one::<String>("url") {
+                config.default_url = Some(url.to_string());
+                config.to_file(&config_path)?;
+                println!("Saved {} as new default_url", url);
+                return Ok(());
+            }
+        }
+
+        let url = if let Some(server_url) = matches.get_one::<String>("url") {
+            if config.default_url.is_none() {
+                config.default_url = Some(server_url.to_string());
+                config.to_file(&config_path)?;
+            }
+            server_url.to_string()
+        } else if let Some(ref server_url) = config.default_url {
+            server_url.to_string()
+        } else {
+            panic!("Server URL is required through --url parameter, AYB_SERVER_URL environment variable, or default_url in {}", config_path.display());
+        };
+
+        let token = matches
+            .get_one::<String>("token")
+            .or(config.authentication.get(&url))
+            .cloned();
+        let client = AybClient {
+            base_url: url.to_string(),
+            api_token: token,
+        };
+        if let Some(matches) = matches.subcommand_matches("create_database") {
```

Comment:
The changes below are just unindenting by one level since we removed a layer of `if` to get the URL above.